internal/terminal/table_test.go

```diff
@@ -0,0 +1,263 @@
+package terminal
+
+import (
+	"fmt"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/fatih/color"
+	"testing"
+)
+
+func TestEmptyTableConstructor(t *testing.T) {
```

Comment:
I think we can combine the `TestEmptyTableConstructor` and `TestTableConstructor` into a single `TestNewTable` test function.

the test table would then look like:

```go
for _, tc := range []struct{
  description string
  header []string
  data []map[string]interface{}
  expectedData []map[string]string
  expectedWidths []map[string]int
}{
  // test cases
} {
  // inside the test here, you could then just assert.Equal(t, tc.expectedData, t.data) and the like
}
```

what this will allow is:
1. the two empty constructor tests you have here are effectively equal, then they could be defined with the above test case struct as simply:

```go
{
  description: "Should return an empty table if no headers are provided",
}
```

(no declarations for `tc.header` nor `tc.data` are needed here, and the equality checks with `expectedData` and `expectedWidths` also defined as `nil` should pass just fine)

the other two test cases you have for `TestTableConstructor` should fit in nicely here as well